torch/_inductor/kernel/flex_attention.py

```diff
@@ -305,6 +305,13 @@ def build_subgraph_buffer(
  """,
 )
 
+def _use_flex_decoding(query):
+    # Decide which kernel to use, return true if use flex decoding kernel. 
+    if query.get_size()[-2] <= 32:
+        return True
+    else: 
+        return False
+
```

Comment:
FIXME: use get_size to determine which kernel to use breaks dynamic shape support for flex_decoding kernel.

Reply:
Can you try ```int(query.get_size())```? I think it would convert symint into int, but probably this would lead to bad heuristic selection if the query seqlen is changed later.

Reply:
https://github.com/pytorch/pytorch/blob/207c2248a881d261ce42566cde7ca25b134fd382/torch/_inductor/sizevars.py#L368-L369

Reply:
Solved using `evaluate_expr`. commit https://github.com/pytorch/pytorch/pull/128678/commits/8301ce374aed91a33612b60f9ae03324133c5d19